float8_experimental/float8_linear.py

```diff
@@ -82,41 +82,47 @@ class NoopFwToFloat8E5M2Bw(torch.autograd.Function):
     def forward(
         ctx,
         tensor,
-        fp8_amax_dL_dY,
-        fp8_amax_history_dL_dY,
-        fp8_scale_dL_dY,
+        fp8_amax_grad_output,
+        fp8_amax_history_grad_output,
+        fp8_scale_grad_output,
         scale_fn_name,
         is_amax_initialized,
         linear_mm_config: LinearMMConfig,
     ):
-        ctx.save_for_backward(fp8_amax_dL_dY, fp8_amax_history_dL_dY, fp8_scale_dL_dY)
+        ctx.save_for_backward(
+            fp8_amax_grad_output, fp8_amax_history_grad_output, fp8_scale_grad_output
+        )
         ctx.scale_fn_name = scale_fn_name
         ctx.is_amax_initialized = is_amax_initialized
         ctx.linear_mm_config = linear_mm_config
         return tensor
 
     @staticmethod
     def backward(ctx, go):
```

Comment:
should we update go? This one has confused me in the past

Reply:
this PR only contains the user-facing part since that's the most crucial.  LOC is already high, so IMO better to rename the non-user facing things in subsequent PRs, to keep things smaller and more reviewable.